x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/aggs/InternalInferenceAggregation.java

```diff
@@ -48,24 +51,44 @@ public InternalAggregation reduce(List<InternalAggregation> aggregations, Reduce
     }
 
     @Override
-    @SuppressWarnings("unchecked")
     public Object getProperty(List<String> path) {
-        Map<String, Object> resultMap = this.inferenceResult.writeResultToMap();
-
-        for (int i=0; i<path.size() -1; i++) {
-            Object value = resultMap.get(path.get(i));
-            if (value == null) {
-                throw new InvalidAggregationPathException("Cannot find an key [" + path.get(i) + "] in " + path);
-            }
-
-            if (value instanceof Map<?, ?>) {
-                resultMap = (Map<String, Object>)value;
+        if (path.isEmpty()) {
+            return this;
+        } else if (path.size() == 1) {
+            String field = path.get(0);
+            if (CommonFields.VALUE.getPreferredName().equals(field)) {
+                if (inferenceResult instanceof ClassificationInferenceResults) {
+                    return ((ClassificationInferenceResults)inferenceResult).transformedPredictedValue();
+                } else if (inferenceResult instanceof SingleValueInferenceResults) {
+                    return ((SingleValueInferenceResults)inferenceResult).value();
+                } else {
+                    return null;
+                }
+            } else if (SingleValueInferenceResults.FEATURE_IMPORTANCE.equals(field)) {
+                if (inferenceResult instanceof SingleValueInferenceResults) {
+                    SingleValueInferenceResults valueResult = (SingleValueInferenceResults) inferenceResult;
+                    return valueResult.getFeatureImportance();
+                } else {
+                    return null;
+                }
+            } else if (ClassificationConfig.DEFAULT_TOP_CLASSES_RESULTS_FIELD.equals(field)) {
```

Comment:
similar to the `feature_importance` path. Could we support getting the top class probability?